migrations/20200624121732_consent.ts

```diff
@@ -0,0 +1,47 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ahan Gupta <ahangupta.96@gmail.com>
+
+ --------------
+ ******/
+import * as Knex from 'knex'
+
+exports.up = async (knex: Knex): Promise<void> => {
+  return await knex.schema.hasTable('Consent').then(function (exists: boolean): Knex.SchemaBuilder | void {
+    if (!exists) {
```

Comment:
A general comment about style: it tends to be easier to read if you just exit early and make the _main thing you want to do_ be at the end of the function, e.g.

```suggestion
   //e.g. take care of negative cases first
    if (exists) {
      return;
    }

    // This leaves room for other conditional handling in the future here
    
    //Then  we can do the thing that we're all here to do
    return knex.schema.createTable(...)

```

Reply:
In Knex example given here: http://knexjs.org/#Schema-createTable is used `if(!exist)` version, but then for the `else` case the returned value is `undefined` in an implicit way. Explicit way always speeds up reading the code. This pattern also allows us to keep if/else nested blocks as flat as possible for complex cases. Let use the proposed by @lewisdaly implicit way.